F841 is triggered by unused variables in tuple unpacking from with statements

When tuple unpacking is used within `as` of a with statement, F841 is raised for unused variables. This is fine if really all variables are unused, but if I need only one, I have not other choice then getting all. This happens only within functions.

```python3
def connect():
    return (None, None)


def test():
    with connect() as (connection, cursor):
        cursor.execute("SELECT * FROM users")

```

using ruff 0.0.135 gives me
 
```
test.py:7:24: F841 Local variable `connection` is assigned to but never used
```

In flake8, this is case works.

--- a/resources/test/fixtures/F841.py
+++ b/resources/test/fixtures/F841.py
@@ -57,3 +57,11 @@ def f5():
 def f6():
     annotations = []
     assert len([annotations for annotations in annotations])
+
+
+def f7():
+    def connect():
+        return None, None
+
+    with connect() as (connection, cursor):
+        cursor.execute("SELECT * FROM users")

--- a/src/check_ast.rs
+++ b/src/check_ast.rs
@@ -6,6 +6,7 @@ use std::path::Path;
 use itertools::Itertools;
 use log::error;
 use rustc_hash::{FxHashMap, FxHashSet};
+use rustpython_ast::Withitem;
 use rustpython_parser::ast::{
     Arg, Arguments, Constant, Excepthandler, ExcepthandlerKind, Expr, ExprContext, ExprKind,
     KeywordData, Operator, Stmt, StmtKind, Suite,
@@ -21,7 +22,7 @@ use crate::ast::types::{
     Binding, BindingContext, BindingKind, ClassScope, FunctionScope, ImportKind, Range, Scope,
     ScopeKind,
 };
-use crate::ast::visitor::{walk_excepthandler, Visitor};
+use crate::ast::visitor::{walk_excepthandler, walk_withitem, Visitor};
 use crate::ast::{helpers, operations, visitor};
 use crate::checks::{Check, CheckCode, CheckKind};
 use crate::docstrings::definition::{Definition, DefinitionKind, Documentable};
@@ -77,6 +78,7 @@ pub struct Checker<'a> {
     in_deferred_string_annotation: bool,
     in_literal: bool,
     in_subscript: bool,
+    in_withitem: bool,
     seen_import_boundary: bool,
     futures_allowed: bool,
     annotations_future_enabled: bool,
@@ -120,6 +122,7 @@ impl<'a> Checker<'a> {
             in_deferred_string_annotation: false,
             in_literal: false,
             in_subscript: false,
+            in_withitem: false,
             seen_import_boundary: false,
             futures_allowed: true,
             annotations_future_enabled: false,
@@ -2080,6 +2083,13 @@ where
 
         self.check_builtin_arg_shadowing(&arg.node.arg, Range::from_located(arg));
     }
+
+    fn visit_withitem(&mut self, withitem: &'b Withitem) {
+        let prev_in_withitem = self.in_withitem;
+        self.in_withitem = true;
+        walk_withitem(self, withitem);
+        self.in_withitem = prev_in_withitem;
+    }
 }
 
 fn try_mark_used(scope: &mut Scope, scope_id: usize, id: &str, expr: &Expr) -> bool {
@@ -2386,7 +2396,7 @@ impl<'a> Checker<'a> {
             return;
         }
 
-        if operations::is_unpacking_assignment(parent) {
+        if self.in_withitem || operations::is_unpacking_assignment(parent) {
             self.add_binding(
                 id,
                 Binding {

